origocli/commands/pipelines/instances.py

```diff
@@ -0,0 +1,195 @@
+import json
+import inquirer
+from inquirer.errors import ValidationError
+from fuzzywuzzy import process
+
+from origo.data.dataset import Dataset
+from origo.pipelines.resources.pipeline_instance import PipelineInstance
+
+from origocli.commands.pipelines.base import BasePipelinesCommand
+from origocli.output import create_output
+
+
+class PipelinesLsInstances(BasePipelinesCommand):
+    """
+    usage: origo pipelines ls-instances --pipeline-arn=<pipeline-arn> [options]
+
+    options:
+      -d --debug
+
+    """
+
+    def default(self):
+        out = create_output(self.opt("format"), "pipelines_instances_config.json")
+        arn = self.opt("pipeline-arn")
+        instances, error = self.sdk.get_pipeline(arn).list_instances()
+        if error:
+            self.log.exception(error)
+            return
+        data = list(map(lambda x: x.__dict__, instances))
+        out.add_rows(data)
+        self.print(f"Instances with arn: {arn}", out)
+
+
+class PipelineInstanceLs(BasePipelinesCommand):
+    """usage:
+      origo pipelines instances ls [(<dataset-id> <version>)] [options]
+
+    options:
+      -d --debug
+      --format=<format>
+    """
+
+    def default(self):
+        out = create_output(self.opt("format"), "pipelines_instances_config.json")
+        dataset_id = self.arg("dataset-id")
+        version = self.arg("version")
+        data = self.sdk.list(PipelineInstance)
+        title = ""
+        if dataset_id and version:
+            data = [
+                instance
+                for instance in data
+                if instance["datasetUri"] == f"output/{dataset_id}/{version}"
+            ]
+            out.add_rows(data)
+            title = "Pipeline instance"
+        else:
```

Comment:
No need for else branch, can move
```
            out.add_rows(data)
            title = "Pipeline instances"
```
outside of if-else block.

Reply:
no, since we are printing after if/else, and we don't want to add all rows in case we hit the first if - then we need to add the add_rows() inside a else statement

Reply:
But you're doing `out.add_rows(data)` in both branches? The `data` will be modified by the `if`-block if needed, so the result will be the same if you move the `add_rows` after the `if`-block. And you're setting title to the same value in both branches, so that can also be moved outside.